### PR TITLE
Rkt Driver Enhancements

### PR DESCRIPTION
**Pull request by dadgar**
* Logs stderr and stdout
* Decoupled exec command from arguments
* Pass environment variables to container
* Evaluate nomad variables in user arguments.

### CODE REVIEW

File: client/driver/rkt.go [hunk 1]
```diff
@@ -78,60 +81,96 @@ func (d *RktDriver) Fingerprint(cfg *config.Config, node *structs.Node) (bool, e
...
-		splitted := strings.Fields(exec_cmd)
-		run_cmd = append(run_cmd, "--exec=", splitted[0], "--")
-		run_cmd = append(run_cmd, splitted[1:]...)
-		run_cmd = append(run_cmd, "---")
+
+	stde, err := os.OpenFile(stderrFilename, os.O_CREATE|os.O_RDWR|os.O_APPEND, 0666)
+	if err != nil {
+		return nil, fmt.Errorf("Error opening file to redirect stderr: %v", err)
 	}
-	acmd := exec.Command(run_cmd[0], run_cmd[1:]...)
```
Comment (achanda) [inline client/driver/rkt.go]: I think we should print exactly how rkt is being run. At least in debug mode.

File: client/driver/rkt.go [hunk 2]
```diff
@@ -78,60 +81,96 @@ func (d *RktDriver) Fingerprint(cfg *config.Config, node *structs.Node) (bool, e
...
-	acmd.Stdout = &aoutBuf
-	acmd.Stderr = &aerrBuf
-	d.logger.Printf("[DEBUG] driver:rkt: starting rkt command: %q", acmd.Args)
```
Comment (achanda) [inline client/driver/rkt.go]: I think we should print exactly how rkt is being run. At least in debug mode.

Comment (dadgar) [inline client/driver/rkt.go]: Added it back :+1:

File: client/driver/rkt.go [hunk 3]
```diff
@@ -78,60 +81,96 @@ func (d *RktDriver) Fingerprint(cfg *config.Config, node *structs.Node) (bool, e
 
 // Run an existing Rkt image.
 func (d *RktDriver) Start(ctx *ExecContext, task *structs.Task) (DriverHandle, error) {
+	// Validate that the config is valid.
 	trust_prefix, ok := task.Config["trust_prefix"]
```
Comment (armon) [inline client/driver/rkt.go]: This should be made optional, the UX is sad otherwise

File: client/driver/rkt.go [hunk 4]
```diff
@@ -78,60 +81,96 @@ func (d *RktDriver) Fingerprint(cfg *config.Config, node *structs.Node) (bool, e
...
 	if !ok || trust_prefix == "" {
 		return nil, fmt.Errorf("Missing trust prefix for rkt")
 	}
 
+	name, ok := task.Config["name"]
```
Comment (armon) [inline client/driver/rkt.go]: Rename this, not clear what the `name` is

File: client/driver/rkt.go [hunk 5]
```diff
@@ -78,60 +81,96 @@ func (d *RktDriver) Fingerprint(cfg *config.Config, node *structs.Node) (bool, e
...
+	envVars := TaskEnvironmentVariables(ctx, task)
+	for k, v := range envVars.Map() {
+		cmd_args = append(cmd_args, fmt.Sprintf("--set-env=%v=%v", k, v))
+	}
+
+	// Append the run command.
+	cmd_args = append(cmd_args, "run", "--mds-register=false", name)
+
+	// Check if the user has overriden the exec command.
+	if exec_cmd, ok := task.Config["exec"]; ok {
```
Comment (armon) [inline client/driver/rkt.go]: Lets rename to `command`

File: client/driver/rkt.go (hunk 3 above)
Comment (achanda) [inline client/driver/rkt.go]: The we would need to have instructions on how to pre-populate trusted keys for `rkt`. Docs: https://coreos.com/rkt/docs/0.5.4/commands.html

File: client/driver/rkt.go (hunk 4 above)
Comment (achanda) [inline client/driver/rkt.go]: Agreed, `name` was a bad choice.

**Pull request merged**